### PR TITLE
Fix property set with subtype

### LINKED ISSUE #904: Cannot set property to subtype
The class `TZDateTime` from `timezone` package implements `DateTime`, but it is not possible to set `DateTime` property on a realm model to a `TZDateTime`. This is due to how `_intoRealmValue` uses a switch on the values `runtimeType` instead of testing with `is`.

### PR DESCRIPTION
**Pull request by nielsenko**
This fixes #904

### CODE REVIEW

File: lib/src/native/realm_core.dart [hunk 1]
```diff
@@ -1997,56 +1997,48 @@ void _intoRealmValue(Object? value, Pointer<realm_value_t> realm_value, Allocato
...
-        if (microseconds < 0 && nanoseconds != 0) {
-          nanoseconds = nanoseconds - _nanosecondsPerMicrosecond * _microsecondsPerSecond;
-        }
-        realm_value.ref.values.timestamp.seconds = seconds;
-        realm_value.ref.values.timestamp.nanoseconds = nanoseconds;
-        realm_value.ref.type = realm_value_type.RLM_TYPE_TIMESTAMP;
-        break;
-      default:
-        throw RealmException("Property type ${value.runtimeType} not supported");
+    if (value is int) {
```
Comment (nirinchev) [inline lib/src/native/realm_core.dart:2000]: If only we had pattern matching ;)

**Review by nirinchev — APPROVED** [review]

File: lib/src/native/realm_core.dart (hunk 1 above)
Comment (blagoev) [inline lib/src/native/realm_core.dart:2000]: The code will work but why are we doing this change. Did the previous version had drawbacks?

File: pubspec.yaml [hunk 2]
```diff
@@ -37,6 +37,7 @@ dev_dependencies:
   json_serializable: ^6.3.1
   lints: ^2.0.0
   test: ^1.14.3
+  timezone: ^0.9.0 # Used for testing only
```
Comment (blagoev) [inline pubspec.yaml:40]:
```suggestion
  timezone: ^0.9.0
```

Comment (blagoev) [reply pubspec.yaml:40]: if we need it as dev_dep then we need it. I suggest we don't have the comment here

File: lib/src/native/realm_core.dart (hunk 1 above)
Comment (blagoev) [reply lib/src/native/realm_core.dart:2000]: ok I understand now

**Review by blagoev — APPROVED** [review]

**Review by blagoev — APPROVED** [review]
No changelog entry

**Review by desistefanova — APPROVED** [review]

**Comment by nielsenko** [comment]
> No changelog entry

Added

**Pull request merged**